WorkerDedicatedRunLoop performs work outside of an autorelease pool

bnham opened the pull request:
#### 8ced70855b0027f1af965cfc723563a1cafedc24
<pre>
WorkerDedicatedRunLoop performs work outside of an autorelease pool
<a href="https://bugs.webkit.org/show_bug.cgi?id=267104">https://bugs.webkit.org/show_bug.cgi?id=267104</a>
<a href="https://rdar.apple.com/119336173">rdar://119336173</a>

Reviewed by Youenn Fablet.

We&apos;ve gotten some os_log_fault logs which seem to indicate that WorkerDedicatedRunLoop is missing
the appropriate calls to push/pop autorelease pools. Fix this by adding an autorelease pool to
`WorkerDedicatedRunLoop::runInMode`.

* Source/WebCore/workers/WorkerRunLoop.cpp:
(WebCore::WorkerDedicatedRunLoop::runInMode):

Canonical link: <a href="https://commits.webkit.org/272700@main">https://commits.webkit.org/272700@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e0e722e0d03f95a8186883487245090858e8c748

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32745 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11500 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34605 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35319 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29584 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33672 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13848 "Built successfully") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8687 "Hash e0e722e0 for PR 22405 does not build (failure)") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/29063 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33188 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9701 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29254 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/8451 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/8591 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/29207 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36656 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29759 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29615 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/34710 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/8708 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/51/builds/8687 "Hash e0e722e0 for PR 22405 does not build (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32577 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10392 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7607 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/9319 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9316 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/workers/WorkerRunLoop.cpp:
```diff
@@ -144,6 +145,7 @@ class RunLoopSetup {
 private:
     WorkerDedicatedRunLoop& m_runLoop;
     IsForDebugging m_isForDebugging { IsForDebugging::No };
+    AutodrainedPool pool;
```
This one might be redundant with the one you add below in WorkerDedicatedRunLoop::runInMode.
I would tend to keep only the one in WorkerDedicatedRunLoop::runInMode if possible.

bnham committed:
WorkerDedicatedRunLoop performs work outside of an autorelease pool
https://bugs.webkit.org/show_bug.cgi?id=267104
rdar://119336173

Reviewed by Youenn Fablet.

We've gotten some os_log_fault logs which seem to indicate that WorkerDedicatedRunLoop is missing
the appropriate calls to push/pop autorelease pools. Fix this by adding an autorelease pool to
`WorkerDedicatedRunLoop::runInMode`.

* Source/WebCore/workers/WorkerRunLoop.cpp:
(WebCore::WorkerDedicatedRunLoop::runInMode):

Canonical link: https://commits.webkit.org/272700@main
